force-app/main/default/classes/RecurrenceRule_TEST.cls

```diff
@@ -210,4 +255,34 @@ public with sharing class RecurrenceRule_TEST {
 
         System.assertEquals(expected, actual);
     }
+
+    @IsTest
+    private static void shouldBuildMonthlyFrequencyWithbyMonthDay() {
+        String expected = 'FREQ=MONTHLY;UNTIL=20200731T000000Z;BYSETPOS=-1;BYMONTHDAY=28,29,30,31';
+
+        RecurrenceRule rule = new RecurrenceRule()
+            .withFrequency('MONTHLY')
+            .withEndDate(Date.newInstance(2020, 7, 31))
+            .withByMonthDays(DateTime.newInstance(2020, 1, 31, 13, 0, 0));
+
+        Test.startTest();
+        String actual = rule.build();
+        Test.stopTest();
+
+        System.assertEquals(expected, actual);
```

Comment:
Do we want to add messages to all of the asserts like we talked about the other day?